### PR TITLE
Zap support

### PR DESCRIPTION
**Pull request by callebtc**
This PR adds zap support to the LNURLp extension. It requires the `nostrclient` ([link](https://github.com/lnbits/nostrclient)) extension to be installed in order to send zap receipts to nostr. Many relays should be added to nostrclient so enough people see the zaps (the zap receipts are not yet sent to the relays in the zap note, see todo). 

Todo:
- [x] Admin settings: set nostr private key for zap server
- [x] Send zap_receipts to relays attached to zap
- [x] Setting to turn off zaps for a specific address
- [x] ~~local `ws` vs `wss` (I need `wss` because I run uvicorn with https)~~

### CODE REVIEW

**Comment by callebtc** [comment]
> Is it okay to generate a new zap pubkey every time LNbits is restarted?
> 
> I'm not saying it isn't, what I am saying is that I don't really understand why zaps are required to be signed by arbitrary key at all, if this is possible.

Yeah I think it's ok, nobody is checking anything anyways (there would have to be a list of certified zap servers otherwise). I've hardcoded it before and I'm planning to add some sort of setting where you can add the pk as an admin user maybe.

Have you tried the PR?

**Comment by Bashy** [comment]
What would be the downside of a user defined privkey for the zapper?

It'd allow to whitelist the zapper pubkey on some private or written restricted relays.

**Comment by callebtc** [comment]
> What would be the downside of a user defined privkey for the zapper?

Nothing in particular, it would just be a setting in the interface.

> It'd allow to whitelist the zapper pubkey on some private or write restricted relays.

That's a good point!

**Comment by prusnak** [comment]
> I've hardcoded it before and I'm planning to add some sort of setting where you can add the pk as an admin user maybe.

Yeah, it makes sense to have the similar setting to `CASHU_PRIVATE_KEY` in the cashu extension to set the Zap private key.

**Comment by talvasconcelos** [comment]
Possible fix #9 ?

File: __init__.py [hunk 1]
```diff
@@ -7,6 +7,28 @@
...
 from lnbits.tasks import catch_everything_and_restart
+from loguru import logger
+
+
+from .nostr.event import Event
+from .nostr.key import PrivateKey, PublicKey
+from environs import Env
+
+
+def generate_keys(private_key: str = ""):
```
Comment (motorina0) [inline __init__.py:18]: this should go in `heplers.py`

**Pull request merged**